Makefile

```diff
@@ -1,9 +1,43 @@
 TOOL_VERSION = 2.1.12
 TEST_DATA = test/simpleTestForGenerator
-
+GIT_AUTHOR = License Publisher (maintained by Gary O'Neall) <gary@sourceauditor.com>
+LICENSE_DATA_REPO_NO_SCHEME = github.com/spdx/license-list-data.git
+LICENSE_DATA_REPO = https://$(LICENSE_DATA_REPO_NO_SCHEME)
+LICENSE_DATA_URL = https://$(GITHUB_TOKEN)@$(LICENSE_DATA_REPO_NO_SCHEME)
+LICENSE_OUTPUT_DIR = .tmp
+VERSION = $(shell git describe --always || echo 'UNKNOWN')
+RELEASE_DATE = $(shell date '+%Y-%m-%d')
+COMMIT_MSG = License list build $(VERSION) using license list publisher $(TOOL_VERSION)
+RELEASE_MSG = Adding release matching the license list XML tag $(VERSION)
+#TODO Change the license data repo to license-list-xml before merging pull request
+	
 .PHONY: validate-canonical-match
-validate-canonical-match: licenseListPublisher-$(TOOL_VERSION).jar-valid resources/licenses-full.json $(TEST_DATA) .tmp
-	java -jar -DLocalFsfFreeJson=true -DlistedLicenseSchema="schema/ListedLicense.xsd" licenseListPublisher-$(TOOL_VERSION).jar LicenseRDFAGenerator src .tmp 1.0 2000-01-01 $(TEST_DATA) expected-warnings
+validate-canonical-match: licenseListPublisher-$(TOOL_VERSION).jar-valid $(TEST_DATA) $(LICENSE_OUTPUT_DIR)
+	java -jar -DLocalFsfFreeJson=false -DlistedLicenseSchema="schema/ListedLicense.xsd" licenseListPublisher-$(TOOL_VERSION).jar LicenseRDFAGenerator src $(LICENSE_OUTPUT_DIR) 1.0 2000-01-01 $(TEST_DATA) expected-warnings
+
+.PHONY: deploy-license-data
+deploy-license-data: licenseListPublisher-$(TOOL_VERSION).jar-valid $(TEST_DATA)
+	rm -rf $(LICENSE_OUTPUT_DIR)
+	git clone $(LICENSE_DATA_URL) $(LICENSE_OUTPUT_DIR) --quiet --depth 1
+	# Clean out the old data directories
+	find "$(LICENSE_OUTPUT_DIR)" -mindepth 1 -maxdepth 1 -name .git -prune -o -type d -exec rm -rf {} \+
+	rm -f $(LICENSE_OUTPUT_DIR)/licenses.md
+	java -jar -DLocalFsfFreeJson=false -DlistedLicenseSchema="schema/ListedLicense.xsd" licenseListPublisher-$(TOOL_VERSION).jar LicenseRDFAGenerator src $(LICENSE_OUTPUT_DIR) $(VERSION) $(RELEASE_DATE) $(TEST_DATA) expected-warnings
+	
+	git -C "$(LICENSE_OUTPUT_DIR)" add -A .
+	git -C "$(LICENSE_OUTPUT_DIR)" commit --author "$(GIT_AUTHOR)" -m "$(COMMIT_MSG)"
+	git -C "$(LICENSE_OUTPUT_DIR)" push origin
+	
+.PHONY: release-license-data
+release-license-data: deploy-license-data
+	if [[ $VERSION =~ .+-g[a-f0-9]{7} ]]
+	then
+		echo Can not release license data - license list version '$VERSION' does not match a release pattern
```

Comment:
I think you want `$(VERSION)` here and a few lines above.  For example:

```console
$ cat Makefile
VERSION = 5

x:
  echo a '$VERSION' b
$ make x
echo a 'ERSION' b
a ERSION b
```

I also think you can drop the single-quotes, because `$(VERSION)` should not contain shell-sensitive characters.

Reply:
You probably also want `echo` -> `@echo` to avoid printing the command, because for `echo` the command largely duplicates the standard output.

Reply:
BTW - this goal hasn't really been tested (it isn't used in the Travis file), so there are probably some other bugs in here.  We could just comment out this section and save the improvements for later.

Reply:
I'll create a separate PR